components/ide-service/pkg/server/server.go

```diff
@@ -417,14 +418,18 @@ rm -rf /tmp/backend-latest
 			}
 		}
 		if warmUpTask != "" {
-			warmUpEncoded, err := json.Marshal([]gitpodapi.TaskConfig{{
+			warmUpEncoded := new(bytes.Buffer)
```

Comment:
@felladrin @andreafalzetti maybe extract it to another PR, approve and deploy that we don't confuse webapp team in this PR

Reply:
https://github.com/gitpod-io/gitpod/pull/15178